### PR TITLE
[risk=low][RW-11689] WIP: check for notebook transfer completion

### PR DESCRIPTION
**Pull request by jmthibault79**
When we duplicate a workspace, the files (.ipynb, .sas, etc) are not necessarily transferred immediately.  We have to check.  In the old notebook-list component, we did so, polling periodically to recheck.

Unfortunately because that component is so different from the new app-files-list component, I can't reuse the logic, and I've been having difficulty getting the polling to work well.  So this is an "MVP" version of the story which only checks once at page load.  I hope to get the polling working soon, but I feel that this more basic version will help users in its current form.

Tested locally by duplicating some workspaces with Jupyter, R, and SAS files.

The first line of text here is copied from notebook-list, but the second line is new to account for the lack of polling.  Please review.

Screenshot 2024-01-22 at 4 33 26 PM

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [x] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/help-sidebar.tsx [hunk 1]
```diff
@@ -574,9 +575,7 @@ export const HelpSidebar = fp.flow(
                   Cromwell Cloud Environment
                 </h3>
                 <BetaBadge
-                  tooltipContent={
-                    'We are regularly improving the Cromwell experience. If you have feedback, reach out to support@researchallofus.org'
-                  }
+                  tooltipContent={`We are regularly improving the Cromwell experience. If you have feedback, reach out to ${SUPPORT_EMAIL}`}
```
Comment (jmthibault79) [inline ui/src/app/components/help-sidebar.tsx:578]: unrelated - I just noticed an opportunity for reuse

File: ui/src/app/pages/appAnalysis/app-files-list.tsx [hunk 2]
```diff
@@ -139,73 +162,98 @@ export const AppFilesList = withCurrentWorkspace()(
...
-            </DataTable>
-          )}
-        </FlexColumn>
+            </div>
+            <div>
+              Copying 1 or more notebooks from another workspace. This may take{' '}
+              <b>a few minutes</b>.
+            </div>
+            <div>
+              Please refresh your browser to check again. If you continue to see
```
Comment (jmthibault79) [inline ui/src/app/pages/appAnalysis/app-files-list.tsx:251]: this text is my invention so I will run it by UX

File: ui/src/app/pages/appAnalysis/app-files-list.tsx [hunk 3]
```diff
@@ -59,19 +63,57 @@ const styles = reactStyles({
...
+    const [isTransferComplete, setIsTransferComplete] =
+      useState<boolean>(undefined);
     const [filesList, setFilesList] = useState<FileDetail[]>();
     const [showNotebookSizeWarningModal, setShowNotebookSizeWarningModal] =
       useState<boolean>(false);
     const [activeNotebookName, setActiveNotebookName] = useState<string>(null);
 
+    const checkTransferComplete = withErrorModal(
+      {
+        title: 'Error Loading Files',
```
Comment (jmthibault79) [inline ui/src/app/pages/appAnalysis/app-files-list.tsx:107]: copied from the other errorModal.  Should I update this title?

File: ui/src/app/pages/appAnalysis/app-files-list.tsx [hunk 4]
```diff
@@ -80,16 +122,20 @@ export const AppFilesList = withCurrentWorkspace()(
       },
       async () => {
         props.showSpinner();
-        listNotebooks(workspace).then(setFilesList);
+        await listNotebooks(workspace).then(setFilesList);
```
Comment (jmthibault79) [inline ui/src/app/pages/appAnalysis/app-files-list.tsx:125]: keep the spinner up while we're loading

**Review by evrii — COMMENTED** [review]
The code all looks good to me, but I am seeing a spinner when I load the analysis page. Is that intended?

https://github.com/all-of-us/workbench/assets/24514630/17d3e253-8409-4de6-8b1b-40b254b88ccf


image

File: ui/src/app/pages/appAnalysis/app-files-list.tsx [hunk 5]
```diff
@@ -59,37 +62,73 @@ const styles = reactStyles({
     paddingTop: 0,
   },
 });
+
 const notebookSizeThreshold = 5 * 1024 * 1024;
-interface AppFilesListProps extends WithSpinnerOverlayProps {
```
Comment (jmthibault79) [inline ui/src/app/pages/appAnalysis/app-files-list.tsx:63]: The spinner-overlay felt pretty intrusive

**Review by evrii — APPROVED** [review]
Looks good to me!

**Pull request merged**